### PR TITLE
add simple bank generator

### PR DESCRIPTION
**Pull request by ahnitz**
Add a simple script to make a direct match stochastic bank using pycbc. It makes some basic cuts and uses the triangle inequality liberally.

### CODE REVIEW

**Comment by ahnitz** [comment]
For those curious, an example command line looks like. Parameters can be arbitrarily added as long as the waveform generator supports it. 
```
pycbc_brute_bank \
--verbose \
--output-file bank.hdf \
--minimal-match 0.97 \
--buffer-length 2 \
--sample-rate 2048 \
--approximant TaylorF2 \
--psd-model aLIGOZeroDetHighPower \
--params mass1 mass2  \
--min 1.35 10.35  \
--max 1.40  1.40  \
--seed 1 \
--tau0-threshold 1.0 \
--low-frequency-cutoff 15.0
```

**Comment by titodalcanton** [comment]
Hi Alex, could you give a bit of context/motivation for the script? Is the goal not having to rely on sbank for production bank generation, or something simpler?

**Comment by ahnitz** [comment]
@titodalcanton This is a script I was using to play around with generating banks for waveforms with new parameters (i.e. non lalsimulation). I've also used it for doing some simple HM studies.

I don't think it's a good replacement for sbank at this point. There is a lot of overlap between what sbank does and the functionality we provide in pycbc, though, so it would be nice at some point to reduce that. I'm happy to go many different routes towards that longer term goal if it would bring more people on board.

File: bin/bank/pycbc_brute_bank [hunk 1]
```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
```
Comment (titodalcanton) [inline bin/bank/pycbc_brute_bank]: Please add a top docstring with the purpose and use it as the description in the ArgumentParser.

File: bin/bank/pycbc_brute_bank [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+import numpy, h5py, logging, argparse, numpy.random, sys
+import pycbc.waveform, pycbc.filter, pycbc.types, pycbc.psd, pycbc.fft, pycbc.pnutils
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--verbose', action='store_true')
+parser.add_argument('--output-file',
```
Comment (titodalcanton) [inline bin/bank/pycbc_brute_bank]: Maybe add a few required=True for obvious args.

File: bin/bank/pycbc_brute_bank [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        self.data = data 
+
+    def pop(self):
+        if len(self.data) == 0:
+            return None
+        l = self.data[-1]
+        self.data = self.data[:-1]
+        return l
+
+class TriangleBank(object):
```
Comment (titodalcanton) [inline bin/bank/pycbc_brute_bank]: TriangleBank may not sound immediately obvious to everyone, can you add a brief docstring?

File: bin/bank/pycbc_brute_bank [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+            # Update where to calculate matches
+            skip_threshold = 1 - (1 - hp.threshold) * 2.0
+            inc.data = inc.data[matches[inc.data] > skip_threshold]
+            
+            if m > hp.threshold:
+                return True
+            if m > mmax:
+                mmax = m     
+
+    def round(self, gen, params, threshold):
```
Comment (titodalcanton) [inline bin/bank/pycbc_brute_bank]: Will this function name ever conflict with the keyword and confuse people?

**Comment by ahnitz** [comment]
@titodalcanton I've include some of your suggestions. I'll merge if the tests pass. I checked and it looks like *all* executables are checked that the help message works except for a few excluded ones.

**Pull request merged**